example-app/SantokuApp/src/framework/authentication/SecureStorageAdapter.ts

```diff
@@ -0,0 +1,78 @@
+import * as SecureStore from 'expo-secure-store';
+
+import {sha256} from '../utilities/crypto';
+
+const STORED_ITEM_KEYS = {
+  ACTIVE_ACCOUNT_ID: 'activeAccountId',
+} as const;
+
+/**
+ * 指定されたアカウントIDを、アクティブなアカウントとしてセキュアストレージに格納します。
+ * セキュアストレージに格納する際のオプションとして、以下を指定します。
+ * ・keychainAccessible: {@link SecureStore.WHEN_UNLOCKED_THIS_DEVICE_ONLY}
+ *
+ * @param accountId アカウントID
+ * @see {@link https://docs.expo.dev/versions/latest/sdk/securestore/#constants}
+ */
+function saveActiveAccountId(accountId: string): Promise<void> {
+  return SecureStore.setItemAsync(STORED_ITEM_KEYS.ACTIVE_ACCOUNT_ID, accountId, {
+    keychainAccessible: SecureStore.WHEN_UNLOCKED_THIS_DEVICE_ONLY,
+  });
+}
+
+/**
+ * 指定されたパスワードをセキュアストレージに格納します。
+ * 指定されたアカウントIDをSHA256でハッシュ化して、その値をキーとしてセキュアストレージに格納します。
+ *
+ * セキュアストレージに格納する際のオプションとして、以下を指定します。
+ * ・keychainAccessible: {@link SecureStore.WHEN_UNLOCKED_THIS_DEVICE_ONLY}
+ * ・keychainService: ハッシュ化したアカウントID
+ *
+ * @param accountId アカウントID
+ * @param password パスワード
+ * @see {@link https://docs.expo.dev/versions/latest/sdk/securestore/#securestoreoptions}
+ */
+async function savePassword(accountId: string, password: string): Promise<void> {
+  // ログインに利用するような項目は平文で保存しないでハッシュ化する
+  const hash = await sha256(accountId);
+  return SecureStore.setItemAsync(hash, password, {
+    keychainAccessible: SecureStore.WHEN_UNLOCKED_THIS_DEVICE_ONLY,
+    keychainService: hash,
+  });
+}
+
+/**
+ * セキュアストレージからアクティブなアカウントのIDを取得します。
+ * @returns セキュアストレージに存在する場合はアクティブなアカウントIDの文字列、存在しない場合はnull
+ */
+function loadActiveAccountId(): Promise<string | null> {
+  return SecureStore.getItemAsync(STORED_ITEM_KEYS.ACTIVE_ACCOUNT_ID);
```

Comment:
これって、ロックかかってないときはエラーが返ってくる？それとも`null`が返ってくる？

Reply:
https://developer.apple.com/documentation/security/keychain_services/keychain_items/restricting_keychain_item_accessibility

> **When Passcode Set**
> If the user hasn’t set a passcode, you can’t store an item with this setting. If the user removes the passcode from a device, any items with this setting are automatically deleted from the keychain. You can only access items with this setting if the device is unlocked. Use this setting if your app only needs access to items while running in the foreground.
> 
> **When Unlocked**
> Items with this setting are only accessible when the device is unlocked. A device without a passcode is considered to always be unlocked. This is the default accessibility when you don’t otherwise specify a setting.